apps/smart-contracts/token/test/PPO.test.ts

```diff
@@ -445,6 +449,402 @@ describe('=> PPO', () => {
+      const user1PPOBalanceBefore = await ppo.balanceOf(user1.address)
+      const user2PPOBalanceBefore = await ppo.balanceOf(user2.address)
+      const { v, r, s } = await getPermitSignature(
+        user1,
+        ppo,
+        spender.address,
+        BigNumber.from(0),
+        deadline
+      )
+
+      await ppo
+        .connect(spender)
+        .transferFromWithPermit(user1.address, user2.address, BigNumber.from(0), deadline, v, r, s)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(user1PPOBalanceBefore)
+      expect(await ppo.balanceOf(user2.address)).to.eq(user2PPOBalanceBefore)
+    })
+
+    it('transfers to recipient if amount < balance', async () => {
+      const user1PPOBalanceBefore = await ppo.balanceOf(user1.address)
+      const user2PPOBalanceBefore = await ppo.balanceOf(user2.address)
+      const { v, r, s } = await getPermitSignature(
+        user1,
+        ppo,
+        spender.address,
+        user1PPOBalanceBefore.sub(1),
+        deadline
+      )
+
+      await ppo
+        .connect(spender)
+        .transferFromWithPermit(
+          user1.address,
+          user2.address,
+          user1PPOBalanceBefore.sub(1),
+          deadline,
+          v,
+          r,
+          s
+        )
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(1)
+      expect(await ppo.balanceOf(user2.address)).to.eq(
+        user2PPOBalanceBefore.add(user1PPOBalanceBefore.sub(1))
+      )
+    })
+
+    it('transfers to recipient if amount = balance', async () => {
+      const user1PPOBalanceBefore = await ppo.balanceOf(user1.address)
+      const user2PPOBalanceBefore = await ppo.balanceOf(user2.address)
+      const { v, r, s } = await getPermitSignature(
+        user1,
+        ppo,
+        spender.address,
+        user1PPOBalanceBefore,
+        deadline
+      )
+
+      await ppo
+        .connect(spender)
+        .transferFromWithPermit(
+          user1.address,
+          user2.address,
+          user1PPOBalanceBefore,
+          deadline,
+          v,
+          r,
+          s
+        )
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(0)
+      expect(await ppo.balanceOf(user2.address)).to.eq(
+        user2PPOBalanceBefore.add(user1PPOBalanceBefore)
+      )
+    })
+
+    it('emits transfer if amount = 0 and recipient is not spender', async () => {
+      const { v, r, s } = await getPermitSignature(
+        user1,
+        ppo,
+        spender.address,
+        BigNumber.from(0),
+        deadline
+      )
+      expect(spender.address).to.not.eq(user2.address)
+      const tx = await ppo
+        .connect(spender)
+        .transferFromWithPermit(user1.address, user2.address, BigNumber.from(0), deadline, v, r, s)
+
+      await expect(tx)
+        .to.emit(ppo, 'Transfer(address,address,uint256)')
+        .withArgs(user1.address, user2.address, 0)
+    })
+
+    it('emits transfer if amount = 0 and recipient is spender', async () => {
```

Comment:
```suggestion
    it('emits transfer if amount = 0', async () => {
```
Can remove the other test case if no security concerns with PR #323 suggestion. Please do this for other cases.